libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/http/DefaultHttpRequestComposer.kt

```diff
@@ -147,22 +150,37 @@ class DefaultHttpRequestComposer(
         operation: Operation<D>,
         customScalarAdapters: CustomScalarAdapters,
         sendApqExtensions: Boolean,
+        sendEnhancedClientAwarenessExtensions: Boolean,
         query: String?,
     ): Map<String, Upload> {
       return composePostParams(
-          writer, operation, customScalarAdapters, query, apqExtensionsWriter(operation.id(), sendApqExtensions)
+          writer, operation, customScalarAdapters, query, extensionsWriter(operation.id(), sendApqExtensions, sendEnhancedClientAwarenessExtensions)
       )
     }
 
-    private fun apqExtensionsWriter(id: String, sendApqExtensions: Boolean): JsonWriter.() -> Unit {
+    private fun extensionsWriter(
+        apqId: String,
+        sendApqExtensions: Boolean,
+        sendEnhancedClientAwarenessExtensions: Boolean,
+    ): JsonWriter.() -> Unit {
+      if (!sendApqExtensions && !sendEnhancedClientAwarenessExtensions) return { }
+
       return {
-        if (sendApqExtensions) {
-          name("extensions")
-          writeObject {
+        name("extensions")
+        writeObject {
+          if (sendApqExtensions) {
             name("persistedQuery")
             writeObject {
               name("version").value(1)
-              name("sha256Hash").value(id)
+              name("sha256Hash").value(apqId)
+            }
+          }
+
+          if (sendEnhancedClientAwarenessExtensions) {
+            name("clientLibrary")
```

Comment:
Do we have a reference that we could link as a comment here for future readers that would like more context for the feature?